homeassistant/components/screenlogic/services.py

```diff
@@ -27,78 +26,88 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-SET_COLOR_MODE_SCHEMA = cv.make_entity_service_schema(
+BASE_SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_CONFIG_ENTRY): selector.ConfigEntrySelector(
+            {
+                "integration": DOMAIN,
+            }
+        )
+    }
+)
+
+SET_COLOR_MODE_SCHEMA = BASE_SERVICE_SCHEMA.extend(
     {
         vol.Required(ATTR_COLOR_MODE): vol.In(SUPPORTED_COLOR_MODES),
-    },
+    }
 )
 
-TURN_ON_SUPER_CHLOR_SCHEMA = cv.make_entity_service_schema(
+TURN_ON_SUPER_CHLOR_SCHEMA = BASE_SERVICE_SCHEMA.extend(
     {
-        vol.Optional(ATTR_RUNTIME, default=24): vol.Clamp(
-            min=MIN_RUNTIME, max=MAX_RUNTIME
+        vol.Optional(ATTR_RUNTIME, default=24): vol.All(
+            vol.Coerce(int), vol.Clamp(min=MIN_RUNTIME, max=MAX_RUNTIME)
         ),
     }
 )
 
 
 @callback
-def async_load_screenlogic_services(hass: HomeAssistant, entry: ConfigEntry):
+def async_load_screenlogic_services(hass: HomeAssistant):
     """Set up services for the ScreenLogic integration."""
 
-    async def extract_screenlogic_config_entry_ids(service_call: ServiceCall):
+    def get_coordinator(
+        service_call: ServiceCall,
+    ) -> ScreenlogicDataUpdateCoordinator:
         if not (
-            screenlogic_entry_ids := [
-                entry_id
-                for entry_id in await async_extract_config_entry_ids(hass, service_call)
-                if (entry := hass.config_entries.async_get_entry(entry_id))
-                and entry.domain == DOMAIN
-            ]
+            coordinator := hass.data[DOMAIN].get(service_call.data[ATTR_CONFIG_ENTRY])
         ):
             raise HomeAssistantError(
                 f"Failed to call service '{service_call.service}'. Config entry for"
-                " target not found"
+                " not found"
```

Comment:
I'd check the state of the config entry instead and raise `ServiceValidationError` if the config entry isn't loaded. Saying "not found" is a bit confusing I think, since the user managed to select the config entry via the selector.

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/